src/apps/external_executor/grpc.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "ccf/odata_error.h"
+#include "ds/serialized.h"
+
+#include <arpa/inet.h>
+#include <variant>
+#include <vector>
+
+namespace ccf::grpc
+{
+  using CompressedFlag = uint8_t;
+  using MessageLength = uint32_t;
+
+  static constexpr size_t message_frame_length =
+    sizeof(CompressedFlag) + sizeof(MessageLength);
+
+  MessageLength read_message_frame(const uint8_t*& data, size_t& size)
+  {
+    auto compressed_flag = serialized::read<CompressedFlag>(data, size);
+    if (compressed_flag >= 1)
+    {
+      throw std::logic_error(fmt::format(
+        "gRPC compressed flag has unexpected value {}", compressed_flag));
```

Comment:
Still think this flag should be clearer so we have an idea to fix when it's hit - "... unexpected value {} - currently only support unencoded gRPC payloads".